test/pre.test.js

```diff
@@ -164,22 +154,20 @@ describe('execPre', function() {
       execed.second = true;
       setTimeout(
         function() {
-          next('error!');
+          next();
```

Comment:
Just to confirm, this does not work anymore because for `next` no argument passing is done and `throw` should be used instead?

Reply:
This is worth making a note of in Mongoose changelog. In the new promise-based implementation, it is simpler to make `next()` errors take precedence over `done()` errors. In async pre hooks (pre hooks that are marked with isAsync, not async functions) `next()` triggers the next hook, so the hook doesn't block subsequent hooks; then `done()` indicates that the hook is done.

Added note here: https://github.com/Automattic/mongoose/pull/15376/commits/e32a1bd1ca0112c3f07264902d65f72512e849d5